dohq_artifactory/admin.py

```diff
@@ -418,3 +418,147 @@ def add_group(self, name, permissions):
     @property
     def repositories(self):
         return [self._artifactory.find_repository_local(x) for x in self._repositories]
+
+class Token(AdminObject):
+    _uri = "security/token"
+
+    def __init__(
+        self,
+        artifactory,
+        username=None,
+        scope=None,
+        expires_in=None,
+        refreshable=None,
+        audience=None,
+        grant_type=None,
+    ):
+        super(Token, self).__init__(artifactory)
+
+        # Either or is optional
+        if not (username or scope):
+            raise TypeError("Require either username or scope as argument")
+
+        if username is None:
+            username = self._artifactory.session.auth[0]
```

Comment:
This part failed for me. Instead, I suggest that one:
`username = self._artifactory.session.auth.username`

Reply:
I suggest fail if `username is None`. The library support many type of Auth, it's hard to support all of them. https://github.com/devopshq/artifactory#authentication